File: backend/src/entities/userLocation.php
<?php

class UserLocation implements jsonSerializable
{
    private $userId;
    private $userName;
    private $latitude;
    private $longitude;
    private $accuracy;
    private $timestamp;

    public function __construct(
        $userId,
        $userName,
        $latitude,
        $longitude,
        $accuracy,
        $timestamp
    ) {
        $this->userId = $userId;
        $this->userName = $userName;
        $this->latitude = $latitude;
        $this->longitude = $longitude;
        $this->accuracy = $accuracy;
        $this->timestamp = $timestamp;
    }

    public function getFirstName()
    {
        return $this->firstName;
    }

    public function getUserTopic()
    {
        return str_replace('@', '.', $this->email);
    }

    public function jsonSerialize()
    {
        return [
        'userId' => $this->userId,
        'userName' => $this->userName,
        'latitude' => $this->latitude,
        'longitude' => $this->longitude,
        'accuracy' => $this->accuracy,
        'timestamp' => $this->timestamp,
        'isCached' => false
        ];
    }
}

File: android/app/src/main/kotlin/pl/kwasow/data/UserLocation.kt
package pl.kwasow.data

import android.location.Location
import kotlinx.serialization.Serializable

@Serializable
data class UserLocation(
    val userId: Int,
    val userName: String,
    val latitude: Double,
    val longitude: Double,
    val accuracy: Float,
    val timestamp: Long,
    val isCached: Boolean,
) {
    constructor(user: MinimalUser, location: Location, isCached: Boolean) : this(
        user.id,
        user.firstName,
        location.latitude,
        location.longitude,
        location.accuracy,
        location.time,
        isCached,
    )
}
